router/internal/rediscloser/read_replica_test.go

```
@@ -0,0 +1,375 @@
+
+	urls := defaultTestRedisClusterURLs
+	if raw := os.Getenv(clusterURLsEnv); raw != "" {
+		urls = strings.Split(raw, ",")
+	}
+	// Copied because the caller appends query parameters to the first entry.
+	urls = append([]string(nil), urls...)
+
+	parsed, err := url.Parse(urls[0])
+	require.NoError(tb, err, "%s does not hold valid URLs", clusterURLsEnv)
+
+	conn, err := net.DialTimeout("tcp", parsed.Host, 500*time.Millisecond)
```

golangci-lint (2.12.2)

[error] net.DialTimeout must not be called. use (*net.Dialer).DialContext with (*net.Dialer).Timeout

(noctx)